File: clients/cpp/NKTgClient.cpp
#include <iostream>

class NKTgClient {
public:
    double x, v, m, dm_dt;

    NKTgClient(double x_, double v_, double m_, double dm_dt_)
        : x(x_), v(v_), m(m_), dm_dt(dm_dt_) {}

    // p = m * v
    double momentum() const {
        return m * v;
    }

    // NKTg1 = x * p
    double nktg1() const {
        return x * momentum(); 
    }

    // NKTg2 = (dm/dt) * p
    double nktg2() const {
        return dm_dt * momentum();
    }
};

File: clients/csharp/src/NKTgClient.cs
namespace NKTgLaw
{
    /// <summary>
    /// Core client for NKTg Law (Nguyen Khanh Tung).
    /// Standardized implementation for universal dynamics.
    /// </summary>
    public class NKTgClient
    {
        public double X { get; }
        public double V { get; }
        public double M { get; }
        public double DmDt { get; }

        public NKTgClient(double x, double v, double m, double dmDt)
        {
            X = x;
            V = v;
            M = m;
            DmDt = dmDt;
        }

        // Returns linear momentum (p = m * v)
        public double Momentum()
        {
            return M * V;
        }

        // Returns NKTg1 (x * p)
        public double NKTg1()
        {
            return X * Momentum();
        }

        // Returns NKTg2 ((dm/dt) * p)
        public double NKTg2()
        {
            return DmDt * Momentum();
        }
    }
}
